modules/hivemq/src/main/java/org/testcontainers/hivemq/HiveMQContainer.java

```diff
@@ -0,0 +1,568 @@
+package org.testcontainers.hivemq;
+
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import org.apache.commons.io.FileUtils;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.event.Level;
+import org.testcontainers.containers.ContainerLaunchException;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.MultiLogMessageWaitStrategy;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class HiveMQContainer extends GenericContainer<HiveMQContainer> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(HiveMQContainer.class);
+
+    private static final DockerImageName DEFAULT_HIVEMQ_EE_IMAGE_NAME = DockerImageName.parse("hivemq/hivemq4");
+    private static final DockerImageName DEFAULT_HIVEMQ_CE_IMAGE_NAME = DockerImageName.parse("hivemq/hivemq-ce");
+
+    private static final int DEBUGGING_PORT = 9000;
+    private static final int MQTT_PORT = 1883;
+    private static final int CONTROL_CENTER_PORT = 8080;
+    @SuppressWarnings("OctalInteger")
+    private static final int MODE = 0777;
+    private static final @NotNull Pattern EXTENSION_ID_PATTERN = Pattern.compile("<id>(.+?)</id>");
+
+    private final @NotNull ConcurrentHashMap<String, CountDownLatch> containerOutputLatches = new ConcurrentHashMap<>();
+    private boolean controlCenterEnabled = false;
+    private boolean debugging = false;
+
+    private final @NotNull LinkedList<String> prepackagedExtensionsToRemove = new LinkedList<>();
```

Comment:
shouldn't it be `Set`? You can use `LinkedHashSet` if you need to preserve the order.